### PR TITLE
Fix CI

### PR DESCRIPTION
**Pull request by cliu123**
##  opendistro-for-elasticsearch/security pull request intake form
_Please provide as much details as possible to get feedback/acceptance on your PR quickly_

 
 1. __Category:__ (_Enhancement, New feature, Bug fix, Test fix, Refactoring, Maintenance, Documentation_)
 Maintenance
 
 
 2. __Github Issue # or road-map entry, if available:__

 3. __Description of changes:__
The PR includes the following changes each of which is necessary to make the CI pass:

- Upgrade to SDK 14 in order to build OpenSearch artifacts
- Build OpenSearch artifacts in CI
- Change `opensearch.version` to consume the latest version of OpenSearch
- Change the classes and methods that were recently renamed in OpenSearch repo
- Fix a UT failure.

 4. __Why these changes are required?__ 
It's necessary to fix CI to restore further discussion and to track the status of the repo.

 5. __What is the old behavior before changes and new behavior after changes?__ (_Please add any example/logs/screen-shot if available_)
No changes in behavior

 6. __Testing done:__ (_Please provide details of testing done: Unit testing, integration testing and manual testing_)
 UT
 
 
 7. __TO-DOs, if any:__ (_Please describe pending items and provide Github issues# for each of them_)

 8. __Is it backport from main branch?__ (_If yes, please add backport PR # and commits #_)

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -36,6 +36,16 @@ jobs:
         key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
         restore-keys: ${{ runner.os }}-m2
 
+    - name: Checkout OpenSearch
+      uses: actions/checkout@v2
+      with:
+        repository: 'opensearch-project/OpenSearch'
```
Comment (palashhedau) [inline .github/workflows/ci.yml:42]: was opendistro-security already removed from this file?

File: src/main/java/com/amazon/opendistroforelasticsearch/security/configuration/ClusterInfoHolder.java [hunk 2]
```diff
@@ -111,14 +112,14 @@ public Boolean hasNode(DiscoveryNode node) {
     }
 
     private static boolean clusterHas6xNodes(ClusterState state) {
-        return state.nodes().getMinNodeVersion().before(Version.V_7_0_0);
```
Comment (palashhedau) [inline src/main/java/com/amazon/opendistroforelasticsearch/security/configuration/ClusterInfoHolder.java:114]: Are we using Version anywhere else in file? If not lets remove it from import

File: src/main/java/com/amazon/opendistroforelasticsearch/security/configuration/ConfigurationLoaderSecurity7.java [hunk 3]
```diff
@@ -131,7 +131,7 @@ public void noData(String id, String type) {
 
                 //when index was created with ES 6 there are no separate tenants. So we load just empty ones.
                 //when index was created with ES 7 and type not "security" (ES 6 type) there are no rolemappings anymore.
-                if(cs.state().metadata().index(securityIndex).getCreationVersion().before(Version.V_7_0_0) || "security".equals(type)) {
```
Comment (palashhedau) [inline src/main/java/com/amazon/opendistroforelasticsearch/security/configuration/ConfigurationLoaderSecurity7.java:134]: same as above

**Review by debjanibnrj — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (cliu123) [reply .github/workflows/ci.yml:42]: It is a newly added task to checkout OpenSearch repo in order to build OpenSearch artifacts in the following step before buliding security plugin on top of it given that OpenSearch artifacts hasn't been available in Maven repo as of now. Security plugin is built in the `Package` step and didn't get removed.

**Review by palashhedau — APPROVED** [review]

File: src/main/java/com/amazon/opendistroforelasticsearch/security/configuration/ConfigurationLoaderSecurity7.java (hunk 3 above)
Comment (cliu123) [reply src/main/java/com/amazon/opendistroforelasticsearch/security/configuration/ConfigurationLoaderSecurity7.java:134]: No, it's not used anywhere else in this file.

File: src/main/java/com/amazon/opendistroforelasticsearch/security/configuration/ClusterInfoHolder.java (hunk 2 above)
Comment (cliu123) [reply src/main/java/com/amazon/opendistroforelasticsearch/security/configuration/ClusterInfoHolder.java:114]: No, it's not used anywhere else in this file.

Comment (cliu123) [reply src/main/java/com/amazon/opendistroforelasticsearch/security/configuration/ClusterInfoHolder.java:114]: Sure. It'll be taken care of in one of future PRs.

**Pull request merged**